rust_snuba/src/strategies/processor.rs

```diff
@@ -315,9 +315,26 @@ fn _validate_schema(
         return Ok(());
     };
 
-    let Err(error) = schema.validate_json(payload) else {
+    let Err(error) = (match schema.schema_type {
+        SchemaType::Protobuf => {
+            if let Err(error) = schema.validate_protobuf(payload) {
+                Err(error)
+            } else {
+                Ok(())
+            }
+        }
+        SchemaType::Json => {
+            if let Err(error) = schema.validate_json(payload) {
+                Err(error)
+            } else {
+                Ok(())
+            }
+        }
+        _ => Err(SchemaError::InvalidType),
+    }) else {
         return Ok(());
     };
+
```

Comment:
nit: this can be simplified to:
```
let Err(error) = match schema.schema_type {
        SchemaType::Protobuf => schema.validate_protobuf(payload),
        SchemaType::Json => schema.validate_json(payload),
        _ => Err(SchemaError::InvalidType),
    } else {
        return Ok(());
    };
```
Inside `match`, you are unwrapping and rewrapping the same error. Instead, return the result directly.

Reply:
No, it can't, because `validate_protobuf` and `validate_json` do not have the same return types, that's why we have to do this weird thing.